### PR TITLE
Support for TIME(p) and TIMESTAMP(p) to SQLAlchemy

### PR DESCRIPTION
**Pull request by hovaesco**
First point from https://github.com/trinodb/trino-python-client/issues/107

### CODE REVIEW

File: trino/sqlalchemy/compiler.py
Comment (hovaesco) [inline trino/sqlalchemy/compiler.py:150]: I'm not really sure whether it's a correct approach. Interval in Trino has a different value and data type. 
For instance:
- data type: `INTERVAL YEAR TO MONTH`
- value: `INTERVAL '3' MONTH`

File: tests/unit/sqlalchemy/test_datatype_parse.py
Comment (andrewdibiasio6) [inline tests/unit/sqlalchemy/test_datatype_parse.py:186]: are you missing timezone here?

File: trino/sqlalchemy/compiler.py
Comment (andrewdibiasio6) [reply trino/sqlalchemy/compiler.py:150]: Is this func supposed to return the data type or value? It seems like it returns a mix of both.

File: trino/sqlalchemy/compiler.py
Comment (dungdm93) [inline trino/sqlalchemy/compiler.py:158]: I prefer build string instead

```python
    def visit_TIMESTAMP(self, type_, **kw):
        datatype = "TIMESTAMP"
        precision = getattr(type_, "precision", None)
        if precision:
            datatype += f"({precision})"
        if getattr(type_, "timezone", False):
            datatype += " WITH TIME ZONE"
        
        return datatype
```

**Review by dungdm93 — CHANGES_REQUESTED** [review]

File: trino/sqlalchemy/datatype.py
Comment (dungdm93) [inline trino/sqlalchemy/datatype.py:218]: I expect this `if` block handle all cases of `time` & `timestamp` datatype:
```python
    if type_name in ("time", "timestamp"):
        type_kwargs: Dict[str, Any] = dict()
        if type_str.endswith("with time zone"):
            type_kwargs["timezone"] = True
        if type_opts is not None:
            type_kwargs["precision"] = int(type_opts)
        return type_class(**type_kwargs)
```

**Review by dungdm93 — CHANGES_REQUESTED** [review]

File: trino/sqlalchemy/compiler.py
Comment (hovaesco) [reply trino/sqlalchemy/compiler.py]: done for both TIME and TIMESTAMP

File: trino/sqlalchemy/datatype.py
Comment (hovaesco) [reply trino/sqlalchemy/datatype.py]: done

**Review by dungdm93 — APPROVED** [review]

**Review by ebyhr — COMMENTED** [review]
Could you please provide simple test script? As you already know, we don't have integration tests for sqlalchemy. I hesitate to merge with confidence.

**Comment by ebyhr** [comment]
@hovaesco Thanks for providing the code. Could you test with `precision=0`? It creates `time(3)` now, but it should be `time(0)`.

File: trino/sqlalchemy/compiler.py
Comment (dungdm93) [inline trino/sqlalchemy/compiler.py:153]: `range(0,12)` mean `[0, 12)`
Trino is support precision up to `12` (picoseconds), [according the docs](https://trino.io/docs/current/language/types.html#time-p).

Comment (hovaesco) [reply trino/sqlalchemy/compiler.py]: ohh.. thanks for catching it

**Comment by andrewdibiasio6** [comment]
LGTM pending ebyhr approval.

**Review by andrewdibiasio6 — APPROVED** [review]

File: trino/sqlalchemy/compiler.py
Comment (dungdm93) [reply trino/sqlalchemy/compiler.py:155]: NPE. LOL

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merge, thanks!

**Pull request merged**
